resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/AmIpFilterSuite.scala

```diff
@@ -162,8 +162,10 @@ class AmIpFilterSuite extends SparkFunSuite {
     assert(!filter.getProxyAddresses.isEmpty)
     // waiting for configuration update
     eventually(timeout(5.seconds), interval(500.millis)) {
-      assertThrows[ServletException] {
+      try {
         filter.getProxyAddresses.isEmpty
+      } catch {
+        case e: ServletException => true
```

Comment:
maybe
```suggestion
        case e: ServletException => // do nothing
        case t => fail()
```

Reply:
It sounds like the intent of this proposed change is to succeed specifically if there is a `ServletException`, but fail for other exception types. I don't think we need code changes to achieve this though. Exceptions different from `ServletException` propagate from the test and cause a test failure.

To confirm this, I just tried introducing a `throw RuntimeException()` in my local copy AmIpFilter.java. As I expected, the test failed.

Reply:
yes, correct, but `true` is unnecessary

Reply:
I have pushed up changes to incorporate this suggestion. Thank you.